src/Type/StringType.php

```diff
@@ -150,13 +151,21 @@ public function isNonEmptyString(): TrinaryLogic
 		return TrinaryLogic::createMaybe();
 	}
 
+	public function isNonFalsyString(): TrinaryLogic
+	{
+		return TrinaryLogic::createMaybe();
+	}
+
 	public function isLiteralString(): TrinaryLogic
 	{
 		return TrinaryLogic::createMaybe();
 	}
 
 	public function tryRemove(Type $typeToRemove): ?Type
 	{
+		if ($typeToRemove instanceof ConstantStringType && $typeToRemove->getValue() === '0') {
+			return TypeCombinator::intersect($this, new AccessoryNonFalsyStringType());
+		}
 		if ($typeToRemove instanceof ConstantStringType && $typeToRemove->getValue() === '') {
```

Comment:
a few lines below there is a `if ($typeToRemove instanceof AccessoryNonEmptyStringType) {` case.

I guess we also need a `if ($typeToRemove instanceof AccessoryNonFalsyStringType) {` case there, but wasn't able yet to come up with a test-case